src/main/java/com/upsolver/datasources/jdbc/querybuilders/DefaultQueryDialect.java

```diff
@@ -245,4 +259,9 @@ public Connection getConnection(String url, java.util.Properties info) throws SQ
     public String getDriverClassName() {
         return null;
     }
+
+    @Override
+    public String getStringValue(ResultSet rs, int index) throws SQLException {
+        return valueGetters.getOrDefault(rs.getMetaData().getColumnType(index), getString).apply(rs, index);
```

Comment:
Consider changing the interface to something like:

```java
public StringValueGetter getStringValueGetter(ResultSet rs, int columnIndex)
```

Then get it once for each column before the read iteration loop, and use `StringValueGetter.read(ResultSet rs, int columnIndex)` multiple times 👍

Reply:
applied